File: shift_overflow.c
#include <stdio.h>
#include <stdlib.h>

int main() 
{
    unsigned char uc = 0x00;
    
    uc = 0x01 << 7;    
    printf("uc << 7 = %02x\n", uc); 

    uc = 0x01 << 8;
    printf("uc << 8 = %02x\n", uc); 
} 

File: shift_overflow.rs
fn main() {
    let mut uc:u8 = 0x00;
    
    uc = 0x01 << 7;
    println!("uc << 7= {}", uc);

    uc = 0x01 << 8;
    println!("uc << 8 = {}", uc);
    
}